synapse/storage/databases/main/devices.py

```diff
@@ -1638,19 +1638,22 @@ async def check_too_many_devices_for_user(self, user_id: str) -> List[str]:
             """
 
             rows = await self.db_pool.execute(
-                "check_too_many_devices_for_user_last_seen", None, sql, (user_id,)
+                "check_too_many_devices_for_user_last_seen",
+                None,
+                sql,
+                user_id,
             )
             if rows:
                 max_last_seen = max(rows[0][0], max_last_seen)
 
         # Fetch the devices to delete.
         sql = """
-            SELECT DISTINCT device_id FROM devices
```

Comment:
You can't do a `DISTINCT ... ORDER BY last_seen` if `last_seen` doesn't appear in the distinct.

The distinct isn't necessary, as there are unique indexes.